src/kagglehub/pandas_datasets.py

```diff
@@ -0,0 +1,150 @@
+import os
+import sqlite3
+from typing import Any, Callable, Optional, Union
+
+# WARNING: This module is intended to be imported only at runtime, with
+# specific error handling to inform users that they need to install the
+# appropriate `pandas-datasets` extras in order to use these methods. Adding
+# any new dependencies here must correspond with an addition to the
+# list of `pandas-datasets` optional dependencies in pyproject.toml.
+import pandas as pd
+
+from kagglehub.datasets import dataset_download
+
+
+# This is a thin wrapper around pd.read_sql_query so we get the connection
+# closing for free after we've produced the DataFrame
+def wrapped_read_sql_query(sql_query: str, path: str) -> pd.DataFrame:
+    conn = sqlite3.connect(path)
```

Comment:
Is it possible to use a with statement here so the resource gets closed regardless of an error?